src/components/buttons/ButtonStyles.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { ButtonSize, ButtonVariety } from './ButtonTypes';
+
+export const ButtonStyled = styled.button`
+  box-sizing: border-box;
+  display: inline-flex;
+  align-items: center;
+  padding: var(--echoes-dimension-size-0) var(--button-padding);
+  height: var(--button-height);
+  min-height: var(--button-height);
+
+  font: var(--echoes-typography-paragraph-default-semi-bold);
+  color: var(--button-color);
+  text-decoration: none;
+
+  background-color: var(--button-background);
+
+  border: var(--button-border);
+  border-radius: var(--echoes-border-radius-400);
+  outline: none;
+
+  cursor: pointer;
+
+  &:focus,
+  &:focus-visible {
+    background-color: var(--button-background-focus);
+
+    outline: var(--echoes-color-focus-default) solid var(--echoes-focus-border-width-default);
+    outline-offset: var(--echoes-focus-border-offset-default);
+  }
+
+  &:hover {
+    background-color: var(--button-background-hover);
+  }
+
+  &:active {
+    background-color: var(--button-background-active);
+  }
+
+  &:disabled,
+  &:disabled:has(:hover, :active, :focus, :focus-visible) {
+    color: var(--echoes-color-text-disabled);
+    background-color: var(--button-background-disabled);
+
+    border: none;
+
+    cursor: not-allowed;
+    pointer-events: none;
+  }
+`;
+
+ButtonStyled.displayName = 'ButtonStyled';
+
+export const ButtonInnerWrapper = styled.span`
```

Comment:
do we want to deal with overflows like we do with dropdown items? Make sure the prefix & suffix are always visible, use '...' on the label?
![image](https://github.com/user-attachments/assets/26ad6330-f99b-4d88-b5d3-76b7ba347b21)